tools/scripts/make_pokedex_enc_platinum.py

```diff
@@ -0,0 +1,369 @@
+            if (category == 1):
+                enc = encounterData["morning"][encData - 2]
+            if (category == 2):
+                enc = encounterData["night"][encData - 2]
+        else:
+            enc = encounterData["land_encounters"][encData]["species"]
+        if (enc == species):
+            absent = False
+        encData += 1
+    return absent
+
+def SearchWater(species, encounterData, absent):
+    encData = 0
+    while ((encData < 5) and absent):
+        enc = encounterData["surf_encounters"][encData]["species"]
+        if (enc == species):
+            absent = False
+        enc = encounterData["old_rod_encounters"][encData]["species"]
+        if (enc == species):
+            absent = False
+        enc = encounterData["good_rod_encounters"][encData]["species"]
+        if (enc == species):
+            absent = False
+        enc = encounterData["super_rod_encounters"][encData]["species"]
+        if (enc == species):
+            absent = False
+        encData += 1
+    return absent
+
+def SearchRadar(species, encounterData, absent):
+    encData = 0
+    while ((encData < 4) and absent):
+        enc = encounterData["radar"][encData]
+        if (enc == species):
+            absent = False
+        encData += 1
+    return absent
+
+def SearchSpecial(species, encounterData, absent):
+    encData = 0
+    if (species == "SPECIES_NONE"):
+        return False
+    while ((encData < len(encounterData["special_encounters"])) and absent):
+        enc = encounterData["special_encounters"][encData]
+        if (enc == species):
+            absent = False
+        encData += 1
+    return absent
+
+def SearchSpecialNatDex(species, encounterData, absent):
+    encData = 0
+    while ((encData < len(encounterData["specialnatdex_encounters"])) and absent):
+        enc = encounterData["specialnatdex_encounters"][encData]
+        if (enc == species):
+            absent = False
+        encData += 1
+    if absent:
+        absent = SearchRadar(species, encounterData, absent)
+    return absent
+
+def CategorySearch(species, encounterData, category):
+    absent = True
+
+    if (category < 3):
+        absent = SearchLand(species, encounterData, absent, category)
+        if absent:
+            absent = SearchWater(species, encounterData, absent)
+    if (category == 3):
+        absent = SearchSpecial(species, encounterData, absent)
+    if (category == 4):
+        absent = SearchSpecialNatDex(species, encounterData, absent)
+
+    return absent
+
+for dexNum, species in enumerate(PokemonSpecies):
+    if species.name in ['SPECIES_BAD_EGG']:
+        continue
+    species = species.name
+
+    for category in range(10):
+        fileNum = NUM_POKEMON * category + dexNum + 4
+        encList = []
+
+        for archiveID in range(183):
+            with open(source_dir / '../../../../field/encounters' / f"{archiveID:03}.json", encoding='utf-8') as encounter_file:
```

Comment:
**polish**: It'd be more robust here for you to build your file paths using the source-dir param passed to your script.

Reply:
Is there a way to point to a certain folder without the path from one to the other?

Reply:
Use `meson.project_source_root()` to get hte roo